src/core/render/compiler.js

```diff
@@ -198,47 +153,22 @@ export class Compiler {
   _initRenderer() {
     const renderer = new marked.Renderer();
     const { linkTarget, linkRel, router, contentBase } = this;
-    const _self = this;
+    // Supports mermaid
     const origin = {};
 
-    /**
-     * Render anchor tag
-     * @link https://github.com/markedjs/marked#overriding-renderer-methods
-     * @param {String} tokens the content tokens
-     * @param {Number} depth Type of heading (h<level> tag)
-     * @returns {String} Heading element
-     */
-    origin.heading = renderer.heading = function ({ tokens, depth }) {
-      const text = this.parser.parseInline(tokens);
-      let { str, config } = getAndRemoveConfig(text);
-      const nextToc = { depth, title: str };
-
-      const { content, ignoreAllSubs, ignoreSubHeading } =
-        getAndRemoveDocsifyIgnoreConfig(str);
-      str = content.trim();
-
-      nextToc.title = removeAtag(str);
-      nextToc.ignoreAllSubs = ignoreAllSubs;
-      nextToc.ignoreSubHeading = ignoreSubHeading;
-      const slug = slugify(config.id || str);
-      const url = router.toURL(router.getCurrentPath(), { id: slug });
-      nextToc.slug = url;
-      _self.toc.push(nextToc);
-
-      // Note: tabindex="-1" allows programmatically focusing on heading
-      // elements after navigation. This is preferred over focusing on the link
-      // within the heading because it matches the focus behavior of screen
-      // readers when navigating page content.
-      return `<h${depth} id="${slug}" tabindex="-1"><a href="${url}" data-id="${slug}" class="anchor"><span>${str}</span></a></h${depth}>`;
-    };
-
+    // renderer customizers
+    origin.heading = headingCompiler({
+      renderer,
+      router,
+      compilerClass: this,
+    });
     origin.code = highlightCodeCompiler({ renderer });
     origin.link = linkCompiler({
       renderer,
       router,
       linkTarget,
       linkRel,
-      compilerClass: _self,
+      compilerClass: this,
```

Comment:
This can just be `compiler` because it is an instance of the Compiler class, not the class itself.